pallets/interest-accrual/src/lib.rs

```diff
@@ -474,4 +517,28 @@ impl<T: Config> InterestAccrual<T::InterestRate, T::Balance, Adjustment<T::Balan
 			.ok_or(ArithmeticError::Underflow)?;
 		Ok(interest_rate_per_sec)
 	}
+
+	fn rates() -> Self::Rates {
+		RateVec(Rates::<T>::get())
+	}
+}
+
+pub struct RateVec<T: Config>(BoundedVec<RateDetailsOf<T>, T::MaxRateCount>);
+
+impl<T: Config> RateCollection<T::InterestRate, T::Balance, T::Balance> for RateVec<T> {
```

Comment:
This forces every pallet that uses interest accrual to be tight-coupled with the interest accrual pallet. I am not sure, we want that. 

Couldn't we return another error that the pallet loans converts?

Reply:
I'm not sure what you mean? `RateVec` and its `RateCollection` impl are only ever exposed to other pallets through the `Rates` type in the impl of the `InterestAccrual` trait. It only relies on the existing parameters of that trait, and only returns `DispatchError` as its error type. It should be totally decoupled?

Reply:
Yes, of course you are right. Not sure where my head was.